psyneulink/core/components/ports/port.py

```diff
@@ -2388,13 +2388,17 @@ def _gen_llvm_function_body(self, ctx, builder, params, state, arg_in, arg_out,
                 name = None
             elif afferent.sender.modulation == OVERRIDE:
                 # Directly store the value in the output array
-                if f_mod_ptr.type != arg_out.type:
-                    assert len(f_mod_ptr.type.pointee) == 1
-                    warnings.warn("Shape mismatch: Overriding modulation should match parameter port output: {} vs. {}".format(
-                                  afferent.defaults.value, self.defaults.value),
-                                  pnlvm.PNLCompilerWarning)
-                    f_mod_ptr = builder.gep(f_mod_ptr, [ctx.int32_ty(0), ctx.int32_ty(0)])
-                builder.store(builder.load(f_mod_ptr), arg_out)
+                try:
```

Comment:
It should not be possible for user to configure things to fail in this place.
so it should be just `assert f_mod_ptr.type == arg_out.type, "message"`.